numpy-stubs/__init__.pyi

```diff
@@ -919,3 +920,69 @@ def argsort(
     kind: Optional[_SortKind] = ...,
     order: Union[None, str, Sequence[str]] = ...,
 ) -> ndarray: ...
+@overload
+def argmax(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: None = ...,
+    out: Optional[ndarray] = ...,
+) -> integer: ...
+@overload
+def argmax(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: int = ...,
+    out: Optional[ndarray] = ...,
+) -> Union[integer, ndarray]: ...
+@overload
+def argmin(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: None = ...,
+    out: Optional[ndarray] = ...,
+) -> integer: ...
+@overload
+def argmin(
+    a: Union[Sequence[ArrayLike], ndarray],
+    axis: int = ...,
+    out: Optional[ndarray] = ...,
+) -> Union[integer, ndarray]: ...
+@overload
+def searchsorted(
+    a: Union[Sequence[ArrayLike], ndarray],
+    v: _Scalar,
+    side: _Side = ...,
+    sorter: Union[None, Sequence[_IntOrBool], ndarray] = ...,  # 1D int array
+) -> integer: ...
+@overload
+def searchsorted(
+    a: Union[Sequence[ArrayLike], ndarray],
+    v: ArrayLike,
+    side: _Side = ...,
+    sorter: Union[None, Sequence[_IntOrBool], ndarray] = ...,  # 1D int array
+) -> ndarray: ...
+def resize(a: ArrayLike, new_shape: _ShapeLike) -> ndarray: ...
+@overload
+def squeeze(a: _ScalarGeneric, axis: Optional[_ShapeLike] = ...) -> _ScalarGeneric: ...
+@overload
+def squeeze(a: ArrayLike, axis: Optional[_ShapeLike] = ...) -> ndarray: ...
+def diagonal(
+    a: Union[Sequence[Sequence[ArrayLike]], ndarray],  # >= 2D array
```

Comment:
I think we are encountering diminishing returns and increasing code complexity with things like `Sequence[Sequence[ArrayLike]]`.